src/utils/strutils.c

```diff
@@ -342,6 +350,42 @@ int strnunquote(char *dest, size_t size, const char *s, int n,
 }
 
 
+/*
+  Like strnunquote(), but reallocates the destination and writes to
+  position `pos`.
+
+  On allocation error, -3 is returned.
+ */
+int strnput_unquote(char **destp, size_t *sizep, size_t pos, const char *s,
```

Comment:
Hard to follow logic. Please document what is the execution logic, or use a standard function to simplify the expression

Reply:
Good suggestion. When adding explanations it became clear that the code both could be simplified and made more efficient by removing an unnecessary call to realloc().